x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/agent_policy_advanced_fields/custom_fields/index.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiDescribedFormGroup, EuiSpacer, EuiCallOut } from '@elastic/eui';
+
+import { FormattedMessage } from '@kbn/i18n-react';
+import React from 'react';
+
+import type {
+  NewAgentPolicy,
+  AgentPolicy,
+  PackagePolicy,
+  PackagePolicyInput,
+} from '../../../../../types';
+
+import { GLOBAL_DATA_TAG_EXCLUDED_INPUTS } from '../../../../../../../../common/constants';
+
+import { GlobalDataTagsTable } from './global_data_tags_table';
+
+interface Props {
+  agentPolicy: Partial<AgentPolicy | NewAgentPolicy>;
+  updateAgentPolicy: (u: Partial<NewAgentPolicy | AgentPolicy>) => void;
+}
+
+export const CustomFields: React.FunctionComponent<Props> = ({
+  agentPolicy,
+  updateAgentPolicy,
+}) => {
+  const isAgentPolicy = (policy: Partial<AgentPolicy | NewAgentPolicy>): policy is AgentPolicy => {
+    return (policy as AgentPolicy).package_policies !== undefined;
+  };
+
+  const findUnsupportedInputs = (
+    policy: Partial<AgentPolicy | NewAgentPolicy>,
+    excludedInputs: Set<string>
+  ): string[] => {
+    if (!isAgentPolicy(policy)) {
+      return [];
+    }
+
+    const found = new Set<string>([]);
+    policy.package_policies?.forEach((p: PackagePolicy) => {
+      p.inputs.forEach((input: PackagePolicyInput) => {
+        if (excludedInputs.has(input.type)) {
+          found.add(input.type);
+        }
+      });
+    });
+    return Array.from(found);
+  };
```

Comment:
Nit: you can probably shorten this a bit if you want, something like:
```js
const findUnsupportedInputs = (policy) => {
  if (!isAgentPolicy(policy)) {
    return [];
  }

  return policy.package_policies.reduce((unsupportedInputs, p) => {
    return [ ...unsupportedInputs, p.inputs
      .filter((input) => Array.from(excludedInputs).includes(input.type))
      .map((input) => input.type)];
  }, []).flat();
};
```

Reply:
This way the final array may end up with duplicate input types if different packages use similar inputs, and I ran into some type problems that I could not resolve either, so I would like to keep the function as is. Deduplication is important in this case as I am using the output of this to construct a warning strings. I can of course handle duplicates, manually, but the set takes care of them for me.